src/util/spanparsing.cpp

```diff
@@ -34,11 +34,11 @@ Span<const char> Expr(Span<const char>& sp)
     int level = 0;
     auto it = sp.begin();
     while (it != sp.end()) {
-        if (*it == '(') {
+        if (*it == '(' || *it == '{') {
```

Comment:
don't we need to trace what type we are in if it is a paren or a {? 

E.g., `(}` would pass here no?

Reply:
I see. 

bikeshed:
I think something like this could be a little clearer:

```c++
// This function finds the end of the current expression and returns pair(expr_span, rest_span)
// trailing comma is included in the expr_span, e.g.
// `<expr A>,<expr B>` would return `<expr A>,` `<expr B>` 
std::pair<Span<const char>, Span<const char>> Expr(const Span<const char>& sp)
{
    size_t left = 0;
    size_t right = 0;
    auto it = sp.begin();
    while (it != sp.end()) {
        switch (*it) {
        case '(':
        case '{':
            ++left;
            break;
        case '}':
        case ')':
            ++right;
            break;
        }
        bool balanced = right == left;
        // completed a Expr by matching nonzero left/right
        if (balanced && right > 0) break;
        // we have a balanced term as a comma separated entity
        if (balanced && *it == ',') break;
        ++it;
    }
    return std::make_pair(sp.first(it-sp.begin), sp.subspan(it-sp.begin()));
}
```

That said it seem peculiar to me that `,` is left on the string -- maybe preferable to trim it here?

Reply:
That's how the current span-based parsing works. I think discussions about improvements to that are interesting but out of scope for this PR.